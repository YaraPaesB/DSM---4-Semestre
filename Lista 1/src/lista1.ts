const login = (login: string, senha: string) => {
    if (login.length > 100 || senha.length > 100) {
        throw new Error("Ultrapassou o limite de caracteres")
    }

    const originalString = `${login}:${senha}`;
    const base64 = Buffer.from(originalString).toString('base64');
    return base64;
}


const validaCpf = () => {

}

export default {login, validaCpf};