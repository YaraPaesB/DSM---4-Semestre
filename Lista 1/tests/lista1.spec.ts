import lista1 from "../src/lista1"


describe("testes lista 1", () => {
    it("deve calcular o base64", () => {
        const login = 'meulogin';
        const senha = 'minhasenha';
        const base64 = Buffer.from(`${login}:${senha}`).toString('base64');
        const resultadoEsperado = base64;
        let result = lista1.login(login, senha);

        expect(result).toBe(resultadoEsperado);
    })
    it("valida numero de caracteres para parametro login", () => {
        let hasError = false
        try {
            const login = 'y'.repeat(200);
            let result = lista1.login(login, 'senha');
            
        } catch (error) {
            hasError = true;
        }
        expect(hasError).toBe(true);
    })
    it.only("valida numero de caracteres para parametro senha", () => {
        const senha = 'y'.repeat(200);
        expect(() => lista1.login('login', senha)).toThrowErrorMatchingSnapshot();
    })
})