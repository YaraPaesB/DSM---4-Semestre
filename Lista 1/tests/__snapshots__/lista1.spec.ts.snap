// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testes lista 1 valida numero de caracteres para parametro senha 1`] = `"Ultrapassou o limite de caracteres"`;
